### PR TITLE
feat(chat): add proxy for theme images (Issue #1158)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**

add proxy for theme images

Issues:

- Issue #1158

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1675.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1675.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/testData/expectedConstants.ts [hunk 1]
```diff
@@ -198,7 +198,8 @@ export const API = {
   addonsHost: '/api/addons',
   chatHost: '/api/chat',
   sessionHost: '/api/auth/session',
-  defaultIconHost: '/api/themes/image?name=default-model',
+  themeUrl: 'api/themes/image',
+  defaultIconHost: () => `/${API.themeUrl}?name=default-model`,
```
Comment (IlyaBondar) [inline apps/chat-e2e/src/testData/expectedConstants.ts]:
```suggestion
  defaultIconHost: () => `/${API.themeUrl}/default-model`,
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1675.nightly-test.deltixhub.io
>E2E tests status: success

**Review by irinakartun — APPROVED** [review]

File: apps/chat-e2e/src/testData/api/iconApiHelper.ts [hunk 2]
```diff
@@ -30,4 +34,18 @@ export class IconApiHelper extends BaseApiHelper {
         .replaceAll(/><\/path>$/g, Tags.closingTag),
     );
   }
+
+  private isAbsoluteUrl = (url: string): boolean => {
```
Comment (Alexander-Kezik) [inline apps/chat-e2e/src/testData/api/iconApiHelper.ts:38]: we already have it in `utils/app/file.ts`

File: apps/chat/src/pages/api/themes/image/[name].ts [hunk 3]
```diff
@@ -56,9 +53,9 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
       return res.status(500).send(errorsMessages.customThemesConfigNotProvided);
     }
 
-    const { name } = req.query as { name: string };
+    const name = req.query.name;
 
-    if (!name) {
+    if (!name || Array.isArray(name)) {
```
Comment (Alexander-Kezik) [inline apps/chat/src/pages/api/themes/image/[name].ts:58]:
```suggestion
    if (!name || !isString(name)) {
```

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**